flask/cli.py

```diff
@@ -120,6 +120,47 @@ def find_default_import_path():
     return app
 
 
+def find_dotenv():
+    here = os.getcwd()
+    while 1:
+        path = os.path.join(here, '.env')
+        if os.path.isfile(path):
+            return path
+        there = os.path.dirname(here)
+        if there == here:
+            break
+        here = there
+
+
+def _set_env(key, value):
+    if PY2:
+        if isinstance(value, text_type):
+            value = value.encode('utf-8')
+    else:
+        key = key.decode('utf-8', 'surrogateescape')
+        if isinstance(value, bytes):
+            value = value.decode('utf-8', 'surrogateescape')
+    os.environ[key] = value
+
+
+def load_dotenv():
+    """Loads a dotenv file for flask."""
+    path = find_dotenv()
+    if path is None:
+        return
+    os.chdir(os.path.dirname(path))
+    with open(path, 'rb') as f:
+        for line in f:
+            line = line.strip()
+            if not line or line[:1] == '#' or '=' not in line:
+                continue
+            key, value = line.split('=', 1)
+            _set_env(key, value)
```

Comment:
at least that node lib i mentioned on irc doesn't - it seems useful that you can set e.g. FLASK_DEBUG=0 on the shell even though you enable it in your .env file by default